x-pack/plugins/actions/server/builtin_action_types/email.ts

```diff
@@ -114,6 +125,7 @@ export type ActionTypeSecretsType = TypeOf<typeof SecretsSchema>;
 const SecretsSchema = schema.object({
   user: schema.nullable(schema.string()),
   password: schema.nullable(schema.string()),
+  clientSecret: schema.nullable(schema.string()),
```

Comment:
I am able to create this connector specifying clientId and tenantId but not clientSecret. Should that throw a validation error?

Reply:
I experienced this too, and in the event log, I see this kind of error:
```
error sending email: {"error":"invalid_client","error_description":"AADSTS7000216: 'client_assertion', 'client_secret' or 'request' is required for the 'client_credentials' grant type.\r\nTrace ID: 00667a51-4334-4b5b-80ad-dc68b2bcab01\r\nCorrelation ID: eecce7b6-0a86-4182-b54f-92196502d457\r\nTimestamp: 2021-09-16 13:28:02Z","error_codes":[7000216],"timestamp":"2021-09-16 13:28:02Z","trace_id":"00667a51-4334-4b5b-80ad-dc68b2bcab01","correlation_id":"eecce7b6-0a86-4182-b54f-92196502d457","error_uri":"https://login.microsoftonline.com/error?code=7000216"}
```

Reply:
I synced with Yulia offline about this and it looks like a larger issue that we will defer.